week3/homework/src/markDone.js

```diff
@@ -1,24 +1,15 @@
 const write = require('./write');
 const read = require('./read');
 
-function markDone(req, res, toDelete) {
+function markDone(req, res, id) {
   read().then(parsedData => {
-    parsedData.map((toDo, index) => {
-      if (toDo.id === req.query.id) {
-        if (toDo.done !== true) {
-          toDo.done = true;
-          const listDone = JSON.stringify(parsedData, null, 2);
-          write(req, res, listDone).then(Response => {
-            res.send('todo has been done.');
-          });
-        }
- else {
-          toDo.done = false;
-          const listDone = JSON.stringify(parsedData, null, 2);
-          write(req, res, listDone).then(Response => {
-            res.send('todo has been not done yet!');
-          });
-        }
+    parsedData.find(toDo => {
```

Comment:
It would be much better to write
```
let toDo = parsedData.find(toDo => toDo.id === id):
if ( typeof toDo === 'undefined' ) throw new Error('no toDo found for id: '+id);
toDo.done = !toDo.done;
const list = JSON.stringify(parsedData, null, 2);
```